src/tools/rustbook/src/main.rs

```diff
@@ -64,10 +72,32 @@ fn main() {
                 }
             };
         },
+        ("linkcheck", Some(sub_matches)) => {
+            if let Err(broken_links) = linkcheck(sub_matches) {
+                eprintln!("Error: {}", broken_links);
+
+                let links = broken_links.downcast_ref::<BrokenLinks>().unwrap().links();
```

Comment:
This will crash if the error is not a `BrokenLinks`. I would propose using `downcast` instead of `downcast_ref`. In the case where it is a `BrokenLinks` we can just print the links (as you already do). Otherwise, we can just print the error as you do on line 77.